src/odemis/util/test/img_test.py

```diff
@@ -1181,8 +1183,822 @@ def test_rotate_img_metadata(self):
+        # size of the DA calculating from the middle point on the 1-axis downwards
+        expected_pos_y_min = pos_y - self.size[-2] * self.md[model.MD_PIXEL_SIZE][1] / 2
+        self.assertEqual(output_pos[1], expected_pos_y_min)
+
+        # check if the max position of the bounding box is the same as half the
+        # size of the DA calculating from the middle point on the 1-axis upwards
+        expected_pos_y_max = pos_y + self.size[-2] * self.md[model.MD_PIXEL_SIZE][1] / 2
+        self.assertEqual(output_pos[3], expected_pos_y_max)
+
+    def test_get_bounding_box_incomplete_metadata(self):
+        """
+        Verify that removing PIXEL_SIZE in the metadate DA will fail and a LookupError exception is raised.
+        """
+        # create a DA with random values
+        new_da = numpy.random.random(self.size)
+        new_da = model.DataArray(new_da, self.md)
+
+        # remove the PIXEL_SIZE key
+        del self.md[model.MD_PIXEL_SIZE]
+
+        # force a LookupError
+        with self.assertRaises(LookupError):
+            img.getBoundingBox(new_da)
+
+    def test_get_bounding_box_wrong_arrayshape(self):
+        """
+        Verify that using a wrong shaped DA will fail and if an IndexError exception is raised.
+        """
+        # create a wrong shaped DA with random values
+        self.size = (4000, 3500)
+        new_da = numpy.random.random(self.size)
+        new_da = model.DataArray(new_da, self.md)
+
+        # force an IndexError due to wrong DA shape
+        with self.assertRaises(IndexError):
+            img.getBoundingBox(new_da)
+
+
+class TestAssembleZCube(unittest.TestCase):
+
+    def setUp(self):
+        # create a 2D size and metadata with necessary items
+        self.size = (3200, 3200)
+        self.md = {
+            model.MD_DIMS: 'YX',
+            model.MD_PIXEL_SIZE: (2e-06, 4e-05),
+            model.MD_POS: (0.2, 0.2)
+        }
+
+    def test_assemble_zcube_normal(self):
+        """
+        Verify that using a 2D shaped image DA's in a list will return the right ZCube assembly.
+        """
+        img_list = []
+        z_list = [0.1, 0.2, 0.3, 0.4, 0.5, 0.5, 0.4, 0.3, 0.2, 0.1]
+
+        # create a list of DA
+        for x in range(len(z_list)):
+            img_list.append(model.DataArray(numpy.random.random_integers(0, 220, self.size), self.md))
+
+        output_DA = img.assembleZCube(img_list, z_list)
+
+        # check if the input DA shapes are different compared to the output DA
+        self.assertNotEqual(img_list[0].ndim, output_DA.ndim)
+        # check if the Z dimension has the right size
+        self.assertEqual(output_DA.shape[0], len(z_list))
+
+    def test_assemble_zcube_3d(self):
+        """
+        Verify that using a 3D shaped image DA's in a list will fail the
+        assembly of a ZCube and see if a ValueError exception is raised.
+        """
+        # adjust the size for 3D DA
+        self.size = (10, 700, 700)
```

Comment:
Why use and change `self.size` while you don't want to change the behaviour of other functions?
=> just use a local `size` .